2 dz/2/main.cpp

```diff
@@ -0,0 +1,35 @@
+/*Реализовать возведение в целую степень (с логарифмической сложностью алгоритма).
+
+*/
+
+#include <iostream>
+
+using namespace std;
+
+double power(double a, int n)
+{
+    double result = 1;
+    while(n > 0)
+    {
+        if (n & 1)
+        {
+            result *= a;
+        }
+        n = n >> 1;
```

Comment:
Note that n is an integer. Bitwise shift operators have undefined behavior when either operand is negative:
https://port70.net/~nsz/c/c11/n1570.html#6.5.7

So, you can remain it, but ensure that n become a non-negative ;)